rust/kernel/delay.rs

```diff
@@ -60,13 +60,10 @@ pub fn coarse_sleep(duration: Duration) {
 ///
 /// This function supports the C side `mdelay`, `udelay`, and `ndelay` functions.
 pub fn coarse_delay(duration: Duration) {
-    let millis = min(
-        duration
-            .as_secs()
-            .saturating_mul(1_000)
-            .saturating_add(duration.subsec_millis() as u64) as core::ffi::c_ulong,
-        core::ffi::c_ulong::MAX,
-    );
+    let millis = duration
+        .as_secs()
+        .saturating_mul(1_000)
+        .saturating_add(duration.subsec_millis() as u64) as core::ffi::c_ulong;
```

Comment:
Does `duration.as_millis().try_into().unwrap_or(u64::MAX)` work or does it give linker errors due to usage of `u128`?

Reply:
Seems like it avoids u128 and works. 
https://github.com/fujita/linux/commit/45c4bcc9a51cc230567855e9cce0fd95b125ca29

Prefer this?